src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/AdyenPaymentServiceProviderPort.java

```diff
@@ -240,8 +247,8 @@ private PurchaseResult handleKlarnaAuthoriseResponse(AdyenCallResult<PaymentsRes
                 }
                 result.setAuthResultKeys(keyTypes);
             }
-        } else {
-            logger.error("Received unexpected resultCode:" + resultCode.toString());
+        } else if(resultCode == PaymentsResponse.ResultCodeEnum.AUTHORISED) {
+            logger.info("Klarna payment response: Authorised");
```

Comment:
Why do we need this logging ?
Will it not be logged as part of PaymentResponse ?

Reply:
Logged for less cluttered response. AdyenCheckoutClient logging is more detailed.